echopype/tests/mask/test_mask.py

```diff
@@ -0,0 +1,222 @@
+import pytest
+
+import numpy as np
+import xarray as xr
+import tempfile
+import os.path
+
+import echopype as ep
+from echopype.mask.api import _check_source_Sv_freq_diff
+
+from typing import List, Union
+
+
+def get_mock_freq_diff_data(n: int, add_chan: bool, add_freq_nom: bool):
```

Comment:
Because of [my comment above](https://github.com/OSOceanAcoustics/echopype/pull/901/files#r1049171963), I think this function should allow creating a dataset with variable number of channels/frequencies, and add a couple tests for selecting two of the X channels/frequencies out of the many channels/frequencies to make sure things run through.

Reply:
That is a good call! I will go ahead and add this functionality.